app/src/main/java/com/babylon/wallet/android/data/repository/BufferedMobileConnectRequestRepository.kt

```diff
@@ -0,0 +1,19 @@
+package com.babylon.wallet.android.data.repository
+
+import com.babylon.wallet.android.domain.model.IncomingMessage
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class BufferedMobileConnectRequestRepository @Inject constructor() {
+
+    private var bufferedRequest: IncomingMessage.IncomingRequest? = null
+
+    fun setBufferedRequest(request: IncomingMessage.IncomingRequest) {
+        bufferedRequest = request
+    }
+
+    fun getBufferedRequest(): IncomingMessage.IncomingRequest? {
+        return bufferedRequest?.also { bufferedRequest = null }
+    }
```

Comment:
actually why not to have this in the `IncomingRequestRepository` ? 🤷🏽‍♂️

Reply:
I was thinking the same thing, more than that, this class just holds some data, so it looks more like a data source than a repository. The repository usually is the component abstracting the data sources, serving as an intermediary between the data sources and the business logic.

Reply:
You are right, will make the change